src/zc/buildout/download.py

```diff
@@ -23,10 +23,20 @@
     from urllib.request import FancyURLopener, URLopener, urlretrieve
     from urllib.parse import urlparse
     from urllib import request
+    import warnings
 
     class PatchedURLopener(FancyURLopener):
         http_error_default = URLopener.http_error_default
 
+        def __init__(self, *args, **kwargs):
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore')
+                # This creates a DeprecationWarning on 3.6:
+                # PatchedURLopener style of invoking requests is deprecated.
+                # Use newer urlopen functions/methods.
+                # Said warning breaks our doctests.
+                super(PatchedURLopener, self).__init__(*args, **kwargs)
+
     request._urlopener = PatchedURLopener()  # Ook! Monkey patch!
```

Comment:
Researching the remaining `ResourceWarnings` has pretty much convinced me that this does nothing in Python 3. I can't find an attribute with that name anywhere in the `urllib` code. There *is* an attribute called `_opener`, which is a different animal; if we try to replace that with this patch, things break badly. 

I'm running tests with this patch completely removed.